File: Day227/uniquePathsInAGrid.java
package Day227;

public class uniquePathsInAGrid {
    public static int uniquePaths(int m, int n) {
        int N = m + n - 2;
        int R = m - 1;
        long answer = 1;
        for (int i=1; i<=R; i++) {
            answer  = answer  * (N - R + i) /i;
        }
        return (int)answer;
    }
    public static void main(String[] args) {
        int m = 3;
        int n = 7;
        System.out.println(uniquePaths(m, n));
    }
}

File: Day227/uniquePathsInAGrid.go
package main

import (
	"fmt"
)

func uniquePaths(m int, n int) int {
    N := n + m  - 2
    R := m - 1
    answer := 1
    for i:=1; i<=R; i++ {
        answer = answer *(N - R + i) / i
    }
    return answer
}

func main() {
	m := 3
	n := 7
	fmt.Println(uniquePaths(m, n))
}